addons/hashes/fnc_hashValues.sqf

```diff
@@ -0,0 +1,25 @@
+#include "script_component.hpp"
+#include "script_hashes.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_hashValues
+
+Description:
+    Returns all values in a Hash.
+
+    See <CBA_fnc_hashCreate>.
+
+Parameters:
+    _hash - Hash to get values for [Array which is a Hash structure]
+
+Returns:
+    Array of all values [Array]
+
+Author:
+    Fusselwurm
+---------------------------------------------------------------------------- */
+SCRIPT(hashValues);
+
+// -----------------------------------------------------------------------------
+params [["_hash", [[], []], [[]]]];
+
++(_hash select HASH_VALUES)
```

Comment:
```suggestion
_hash select HASH_VALUES
```
`select` always creates a copy

Reply:
`ARRAY select NUMBER` returns a reference. For example:
```sqf
private _a = [[1], []]; 
private _b = _a select 0; 
_b pushBack 2; 
_a // -> [[1,2],[]]
```

Reply:
ok, I removed the changes again :sweat_smile:  `+(x select y)` it is!

Reply:
Consider a shallow copy:

``` 

[] + (_hash select HASH_VALUES

```
which will keep value array references as is what I would expect.

Reply:
oh no I'm an idiot; I was thinking of `select {}`